docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
+
+The above sequence diagram shows the operation of how the view orderline method will be called.
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will then match the command keyword `view`
+3. The Regex Class will then be call to match the rest of the user's input to find the id required for retrieving information of the inverntory.
+4. The User Interface Class will call the viewGood() method from the Commands Class
+5. This method will retrieve information of an inventory item by searching through the userOrderlines array list for the corresponding id.
+6. The information retrieved will then be formatted and returned to screen for the user the see the information.
+
+### Add Goods Method
+#### Description
+Add Goods belongs as part of the Commands Class. It is used to add a Good Object into the Collection of Goods Objects currently in the Warehouse.
+
+#### Operation
+![Regex Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/addGood.puml)
+The above sequence diagram shows the operation of how the add goods method will be called. 
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will then match the command keyword `add`
+3. The Regex Class will then be call to match the rest of the user's input to find the necessary values required for the Good Class
+4. Afterwards, the User Interface class will call addGoods() method from the Commands Class
+5. This method will then call the constructor of the Good class, creating a new instance of the Good Object
+6. The addGood() method will then add the Good to the Collections storing the Good Objects currently in the WareHouse
+
+For more examples of how a user can use a command, refer to the [UserGuide](/UserGuide.md)
+
+### Remove Goods Method
+#### Description
+Remove Goods belongs as part of the Commands Class. It is used to remove a certain amount of goods from the inventory.
+#### Operation
+![removeGood diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/removeGood.puml)
+The above sequence diagram shows the operation of how the add goods method will be called.
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will then match the command keyword `remove`
+3. The Regex Class will then be called to match the rest of the user's input to find the values required to remove goods from the inventory.
+4. Afterwards, the User Interface class will call removeGood() method from the Commands Class
+5. This method will then reduce the quantity of a type of goods if the quantity input is not larger than the existing quantity. If the quantity input is the same as the existing quantity, the goods object will be removed from the inventory.
+6. The UI will show a message of format
```

Comment:
Consider including more details on what methods are called and what types of objects are returned by each method, to make the implementation of this program clearer.